hulks/base.py

```diff
@@ -26,8 +26,13 @@ def handle(self, args=None):
         return int(not retval)
 
     def lines_iterator(self, filename):
-        with open(filename) as fp:
-            for line_number, line in enumerate(fp.readlines(), 1):
-                # heuristic, so we dont need to handle all "comment" syntax accross languages
-                if ' noqa' not in line.lower():
-                    yield line_number, line
+        try:
+            with open(filename) as fp:
+                for line_number, line in enumerate(fp.readlines(), 1):
+                    # heuristic, so we dont need to handle all "comment" syntax accross languages
+                    if ' noqa' not in line.lower():
+                        yield line_number, line
+        except UnicodeDecodeError as error:
```

Comment:
acho melhor deixar dentro do try..except somente as linhas que podem levantar a exceção capturada para o código ficar mais claro e simples, sugiro algo como:

```python

def lines_iterator(...):
    with open(filename) as fp:
        try:
            lines = fp.readlines()
        except UnicodeDecodeError as error:
            ...

    for line_number, line in enumerate(lines):
        ...
```

Reply:
Actually, now that you say that  - the code should not be using `readlines` to start with.
I think I will fix my mocking class to work if the code is changed to iterate over the file object itself as well as with readlines, though.

http://stupidpythonideas.blogspot.com/2013/06/readlines-considered-silly.html

otherwise, I agree on narrowing the lines inside the try: block. The 'pretty' thing would be to have an explcit iterator with the try - except block, and have the `for` in lines_iterator use that.  But I think it  might be  overkill, but what is your say?

```
def unicode_error_catcher(iterator, data=''):
      try:
            yield from iterator
      except UnicodeDecodeError:
            # append data to error message
            ...

def lines_iterator(...):
    ...
    for line_number, line in unicore_error_catcher(fp, filename):
        ...
```

Reply:
Agreed, I think in another PR we should use something better than `readlines`